run/gust.ngpu8.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+#PBS -N cm1
+#PBS -l select=2:ncpus=4:mpiprocs=4:ngpus=4:mem=100gb
+#PBS -l walltime=06:00:00
+#PBS -j oe
+#PBS -q main@gusched01
+#PBS -A UNDM0006
+
+#export PALS_NRANKS=8
+#export PALS_PPN=4
+#export PALS_DEPTH=1
+#export PALS_CPU_BIND=depth
+#export MPICH_GPU_SUPPORT_ENABLED=1
+#mpiexec ./gpu.exe --namelist namelist.baseline.dx=2.5m.short-512 >& ngpus=8.dx=2.5m.short-512.log
+
+#export PALS_DEPTH=1
+#export PALS_CPU_BIND=depth
+
+module purge
+module load nvhpc cuda cray-mpich
+
+export MPICH_GPU_SUPPORT_ENABLED=1
+#export CUDA_VISIBLE_DEVICES=$PMI_LOCAL_RANK
+export MPICH_GPU_MANAGED_MEMORY_SUPPORT_ENABLED=1
+#export NVCOMPILER_ACC_NOTIFY=16
+env 
+#mpiexec -n 8 -ppn 4 ./setdev.sh ./cm1.exe --namelist namelist.figureE.gpu >& gpu=8.figureE.512x512.log
+mpiexec -n 8 -ppn 4 ./setdev.sh ./cpu.exe --namelist namelist.figureD.gpu >& gpu=8.figureD.256.log
```

Comment:
Couple of comments:
- We may want to use the global script `get_local_rank` on Gust rather than a self-defined script here.
- Change `cpu.exe` to `gpu.exe`.
- Remove the `=` sign between `gpu` and `8`. Otherwise there is an annoying single quote in the output filename.

Reply:
I was aware of this script on 10/13 so probably it was later than when you were testing CM1 on Gust. But since we are mapping one MPI rank per GPU, staying with your script may be ok.